### DISCUSSION TITLE
Ranks not properly updating when overriding a lazer score on stable

### DISCUSSION THREAD

**Discussion by mooshiros** [ppy/osu#29150] [Q&A] [answered]
On [Cold Weather [Hard]](https://osu.ppy.sh/beatmapsets/1280001#osu/2662771) two days ago I set a gold S rank on the daily challenge (3.4 million lazer classic score). I proceeded to override it with a HDDT SS on stable (5 million scorev1, 3.8 million lazer classic score). Despite this, my profile still displays a gold S rank that it shouldn't and one less silver SS than it should. I then proceeded to go back to lazer and set a HD S rank off the daily challenge (3.5 million lazer classic score) and a NM SS on the daily challenge (3.4 million lazer classic score), and yet my profile still displays a gold S rank and one less silver SS than it should. Is there a way to fix this?

Edit 2024-07-31: I confirmed that this is a general issue with lazer rather than just the daily challenge (probably should have done this earlier) by setting a nomod fc on [Harumachi Clover [Nino's Insane]](https://osu.ppy.sh/beatmapsets/842412#osu/1762730) on lazer then setting a better hd fc on stable. My profile still displays an extra gold S that it shouldn't, and my silver S count did not increment. Feel like I probably should have made this an issue rather than a discussion but oh well.

![image](https://github.com/user-attachments/assets/178b4d2e-a08f-4e0a-8ad6-d3f414d8a4a5)
![image](https://github.com/user-attachments/assets/413128a2-ac8c-4e00-8edf-6172b3bef8bd)
![image](https://github.com/user-attachments/assets/cda290fc-d43d-4476-877d-64cb01bcb270)
![image](https://github.com/user-attachments/assets/e1c08ee7-f385-440f-b064-d21e32222719)
![image](https://github.com/user-attachments/assets/da16c97c-c4d9-4482-8987-8cab3430f32c)

**Comment by peppy** [comment, marked as answer]
Tracking at https://github.com/ppy/osu/issues/29506.
